src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -272,7 +262,7 @@ object DepositHandler {
     }
 
     val fetchItems = getFetchTxt(bagDir).map(_.asScala).getOrElse(Seq())
-    val (fetchItemsInBagStore, itemsToResolve) = fetchItems.partition(bs.isBagStoreAware && _.getUrl.startsWith(bs.baseUrl))
+    val (fetchItemsInBagStore, itemsToResolve) = fetchItems.partition(bs.nonEmpty && _.getUrl.startsWith(bs.get.baseUrl))
```

Comment:
Well, you only need this partition if and only if the bagstore is actually configured. If not, you can just pass on the complete list. Saves you an extra iteration through the list. See https://github.com/vesaakerman/easy-sword2/pull/4/files#diff-ad17b49d50f1b93808fb326e8f79d0d8R262 for my refactoring.